slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,370 @@
+
+-   **impute**: Does the model struggle with missing data?
+
+-   **decorrelate**: Does the model struggle with correlated predictors?
+
+-   **normalize**: Should predictors be centered and scaled?
+
+-   **transform**: Is it helpful to transform predictors to be more symmetric?
+
+::: footer
+<https://www.tmwr.org/pre-proc-table.html>
+:::
+
+## Two types of preprocessing
+
+![](images/fe_venn.svg){fig-align="center"}
+
+## Two types of preprocessing
+
+![](images/fe_venn_info.svg){fig-align="center"}
+
+## General definitions
+
+<br>
+
+* _Data preprocessing_ are the steps that you take to make your model **successful**
+* _Feature engineering_ are what you do to the original predictors to make the model do the **least work** to perform great 
+
+## Working with dates
+
+This data set originally contained a datetime variable. If given as a raw predictor, it is converted to an integer.
+
+It can be re-encoded as:
+
+* Days since a reference date
+* Day of the week
+* Month
+* Year
+* Leap year
+* Indicators for holidays
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute bottom="0" left="0" width="150" height="150"}
+
+*What other transformations could be do with the raw time variable?*
+
+<br>
+
+<br>
+
+Remember that the transformations are tied to the specific modeling problem
+
+```{r}
+#| echo: false
+countdown(minutes = 3, id = "datetime-split")
+```
+
+## two types of transformations
+
+::: columns
+::: {.column width="50%"}
+
+### static
+
+- sqrt, log, inverse
+- dummies with known levels
+- date time extractions
+
+:::
+
+::: {.column width="50%"}
+
+### trained
+
+- centering & scaling
+- Imputation
+- PCA
+- anything for unknown levels
+
+:::
+
+:::
+
+::: fragment
+Trained methods needs to calculate **sufficient information** to be applied again
+:::
+
+## recipes package
+
+extensible framework for pipeable sequences of feature engineering steps provides preprocessing tools to be applied to data
+
+::: {.incremental .highlight-last}
+- Modular + extensible
+- pipeable
+- Deferred evaluation
+- Isolates test data from training data
+- Can do things formulas can't
+:::
+
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>%  
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+[recipe(tip ~ ., data = taxi_train)]{style="color: #CA225E;"} %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>%  
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+<br>
+
+Start by calling `recipe()` to denote the data source and variables used
```

Comment:
What about moving this sentence above the code block and then gradually adding the other two sentences (on steps and selectors) below so that it's all visible at the end?

Reply:
The reason why I put it under, was so the recipes code would occupy the same space in the slide, regardless of how much text we put around it